internal/service/credentials/resource_credential_issuance_rule.go

```diff
@@ -0,0 +1,842 @@
+						Optional:    true,
+						Validators: []validator.Set{
+							setvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("group_ids")),
+							setvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("scim")),
+						},
+					},
+
+					"scim": schema.StringAttribute{
+						Description: "A SCIM query that selects users to which the credential issuance rule applies.",
+						Optional:    true,
+						Validators: []validator.String{
+							stringvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("group_ids")),
+							stringvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("population_ids")),
+						},
+					},
+				},
+			},
+
+			"automation": schema.SingleNestedAttribute{
+				Description: "Contains a list of actions, as key names, and the update method for each action.",
+				Required:    true,
+				Attributes: map[string]schema.Attribute{
+					"issue": schema.StringAttribute{
+						Description:         automationIssueDescription.Description,
+						MarkdownDescription: automationIssueDescription.MarkdownDescription,
+						Required:            true,
+					},
+					"revoke": schema.StringAttribute{
+						Description:         automationRevokeDescription.Description,
+						MarkdownDescription: automationRevokeDescription.MarkdownDescription,
+						Required:            true,
+					},
+					"update": schema.StringAttribute{
+						Description:         automationUpdateDescription.Description,
+						MarkdownDescription: automationUpdateDescription.MarkdownDescription,
+						Required:            true,
+					},
+				},
+			},
+
+			"notification": schema.SingleNestedAttribute{
+				Description: "Contains notification information. When this property is supplied, the information within is used to create a custom notification.",
+				Optional:    true,
+				Attributes: map[string]schema.Attribute{
+					"methods": schema.SetAttribute{
+						ElementType:         types.StringType,
+						Description:         notificationMethodsDescription.Description,
+						MarkdownDescription: notificationMethodsDescription.MarkdownDescription,
+						Optional:            true,
+						Validators: []validator.Set{
+							setvalidator.ValueStringsAre(
+								stringvalidator.OneOf(
+									string(credentials.ENUMCREDENTIALISSUANCERULENOTIFICATIONMETHOD_EMAIL),
+									string(credentials.ENUMCREDENTIALISSUANCERULENOTIFICATIONMETHOD_SMS),
+								),
+							),
+						},
+					},
+					"template": schema.SingleNestedAttribute{
+						Description: "Contains template parameters.",
+						Optional:    true,
+						Attributes: map[string]schema.Attribute{
+							"locale": schema.StringAttribute{
+								Description:         notificationTemplateLocaleDescription.Description,
+								MarkdownDescription: notificationTemplateLocaleDescription.MarkdownDescription,
+								Optional:            true,
```

Comment:
With this schema setup, I think someone can write:
```
template = {}
```
which can cause problems later.  Should `locale` or `variant` be required when setting the `template = {}` nested object?

Reply:
This will functionally work but it probably won't render the docs correctly.  It'll be better to set line 304 and 311 as `Required: true`